library/private/options_tools.h.in

```diff
@@ -23,6 +24,32 @@ std::string trim(std::string_view strv)
   return std::string(strv);
 }
 
+//----------------------------------------------------------------------------
+/** Call `std::stod` and throw `std::invalid_argument` if the whole string has not been consumed. */
+double parse_double(const std::string& str)
```

Comment:
```suggestion
double stodStrict(const std::string& str)
```
better name? not as readable but more explicit

Reply:
not a big fan of this tbh, I like the symmetry of having one method by type, named logically

Reply:
the template version throws "user friendly" exception mentioning the type name, so if we can't really use `double parse()` from within `ratio_t parse()`

Reply:
got it, in that case I prefer `stodStrict` but maybe @Meakk have inputs here.

Reply:
`stodStrict` or `stodThrow` or `stodSafe` are all fine

Reply:
`stodStrict` or `stodThrow` or `stodSafe` are all fine